shm/src/common.cpp

```diff
@@ -0,0 +1,53 @@
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+
+#include "common.hpp"
+
+/*static*/ /*public*/
+//-----------------------------------------------------------------------------------
+// <summary> This method prints out vector of cargo content
+//-----------------------------------------------------------------------------------
```

Comment:
Why you use this comment for each function? If you want document your code, please do this in .hpp and apply for the whole project :)